src/UI/src/shared/components/useRoles.ts

```diff
@@ -0,0 +1,18 @@
+import { getClaims } from "./getClaims";
+import { useEffect, useState } from "react";
+
+export const useRoles = () => {
```

Comment:
I think you can make this easier. You anyway need to check 'Magnet.Read' everytime you load the application. This means you can make the check outside of this function. 
e.g. something like
```ts
let hasAccess :boolean? = undefined;
const promise =  getClaims().then((response) => {
       hasAccess =  response.claims.find((c) => c.value === 'Magnet.Read')
     });

export const useHasAccess = () => {
  if(hasAccess === undefined) { 
      throw promise; // catch it with suspense
  }
  return hasAccess;
}
```